app/main.py

```diff
@@ -1 +1,17 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import VaccineError, NotWearingMaskError
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    masks_to_buy = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            return "All friends should be vaccinated"
```

Comment:
The current logic stops checking friends after encountering a `VaccineError`. If the goal is to check all friends, consider accumulating errors and reporting them after processing all entries. This will ensure that all friends are checked, and the function can provide a comprehensive report.